### ISSUE TITLE
Introduce [+help+] placeholder for Summarize Posts

### ISSUE THREAD

**Issue by GoogleCodeExporter** [viktorthang/wordpress-custom-content-type-manager#382]
```
What functionality do you want to see?

It'd be helpful for shortcode users if there were a [+help+] placeholder that 
would display all available placeholders.

Also, it'd be helpful if there were a way to tie into output filters... sigh... 
this is a wormhole, but of COURSE people will want to convert attachment IDs 
into a url or something.  Maybe something like:

[+my_image:get_post:guid+]

Ugh, I just threw up a little with that... it's slipping down the wormhole.  

And then I could do a search for ":get_post:" and that would trigger some 
stuff...
```

Original issue reported on code.google.com by `ever...@fireproofsocks.com` on 30 May 2012 at 3:32

**Comment by GoogleCodeExporter** [comment]
```
I can confirm that converting attachment into ids to urls would be helpful.  am 
already simulating this with ugly hackish php code in more than one site...
```

Original comment by `mopto...@gmail.com` on 11 Jun 2012 at 8:26

**Comment by GoogleCodeExporter** [comment]
```
Maybe the easiest way to handle this would be to allow .php variations of the 
.tpl formatting strings.  I know that opens up pandora's box with the potential 
of sloppy  code (exhibit A: most WordPress templates), but I don't think it 
makes sense to come up with a whole parsing engine just to allow some extra 
functionality to the tpl values.  

So what I think might be a good balance here would be the possibility to use 
EITHER a .tpl OR a .php version of the template files.  The .tpl files have the 
advantage that they can be reused, cached, will be faster, and will not crash 
or be hacked. 

The  .php variants would have the ability to be customized as needed (e.g. look 
up links when given a post or attachment ID).  They would be included via 
include() and some type of extract() function so the variables/values would be 
available to the included file.

Perhaps there could be a global setting to enable/disable the .php versions of 
the files.  The precedent could be to give priority to the .php files, thus if 
you had formatting_string.php and formatting_string.tpl both, the 
formatting_string.php would get included and the formatting_string.tpl would be 
ignored.
```

Original comment by `ever...@fireproofsocks.com` on 12 Jun 2012 at 8:07

**Comment by GoogleCodeExporter** [comment]
```
See also issue #376.

The only issue with the .php over the .tpls is in the shortcodes... what are 
they to do?  Adding PHP to the shortcodes would be a mess... that's a case 
where having some additional parsing would make some sense.  Maybe nested 
parsing in the shortcodes?
```

Original comment by `ever...@fireproofsocks.com` on 20 Jun 2012 at 4:00

**Comment by GoogleCodeExporter** [comment]
```
The [+help+] placeholder is now available in 0.9.6.1-dev.  See issue #376 for 
some discussion on how the parser needs to work... it's just a matter of 
settling on a syntax.  I've opted for enhancing the parser because that would 
guarantee that the placeholders in the shortcodes would work, whereas the 
possibility of using .php files would not work with the shortcodes... not to 
mention that doing .php files would open up a can of worms as far as caching 
and security is concerned.  Users can write their own output filters and modify 
content that way if needed.
```

Original comment by `ever...@fireproofsocks.com` on 27 Jun 2012 at 5:20
- Changed state: **Fixed**
- Added labels: **Milestone-Release0.9.7**
